cooperlees/monitord
# Add Service Counters

- Add handy per unit counters
  - Refer to README changes to see counters
- Service name needs to be added into `[Service]` config section to enable
- Add how to code gen into Development section of README
- Update docs with new config + JSON output

Test:

- Ensure tests stay passing
- Adjust fixtures + tests to incorporate 15 new keys for 1 service
- Run locally and see stats for chronyd + sshd
  - `cargo run -- -c monitord.conf -vv`
```json
{
  "monitord.services.chronyd.service.active_enter_timestamp": 1683556542382710,
  "monitord.services.chronyd.service.active_exit_timestamp": 0,
  "monitord.services.chronyd.service.cpuusage_nsec": 328951000,
  "monitord.services.chronyd.service.inactive_exit_timestamp": 1683556541360626,
  "monitord.services.chronyd.service.ioread_bytes": 18446744073709551615,
  "monitord.services.chronyd.service.ioread_operations": 18446744073709551615,
  "monitord.services.chronyd.service.memory_available": 18446744073709551615,
  "monitord.services.chronyd.service.memory_current": 5214208,
  "monitord.services.chronyd.service.nrestarts": 0,
  "monitord.services.chronyd.service.restart_usec": 100000,
  "monitord.services.chronyd.service.state_change_timestamp": 1683556542382710,
  "monitord.services.chronyd.service.status_errno": 0,
  "monitord.services.chronyd.service.tasks_current": 1,
  "monitord.services.chronyd.service.timeout_clean_usec": 18446744073709551615,
  "monitord.services.chronyd.service.watchdog_usec": 0,
  "monitord.services.sshd.service.active_enter_timestamp": 1683556543537641,
  "monitord.services.sshd.service.active_exit_timestamp": 0,
  "monitord.services.sshd.service.cpuusage_nsec": 34009000,
  "monitord.services.sshd.service.inactive_exit_timestamp": 1683556543429574,
  "monitord.services.sshd.service.ioread_bytes": 18446744073709551615,
  "monitord.services.sshd.service.ioread_operations": 18446744073709551615,
  "monitord.services.sshd.service.memory_available": 18446744073709551615,
  "monitord.services.sshd.service.memory_current": 2338816,
  "monitord.services.sshd.service.nrestarts": 0,
  "monitord.services.sshd.service.restart_usec": 42000000,
  "monitord.services.sshd.service.state_change_timestamp": 1683556543537641,
  "monitord.services.sshd.service.status_errno": 0,
  "monitord.services.sshd.service.tasks_current": 1,
  "monitord.services.sshd.service.timeout_clean_usec": 18446744073709551615,
  "monitord.services.sshd.service.watchdog_usec": 0,
}
```
